### PR TITLE
Wait until the nodes are ready

### PR DESCRIPTION
**Pull request by ebattat**
## Type of change
Note: Fill **x** in []
- [ ] bug
- [x] enhancement
- [ ] documentation
- [ ] dependencies

## Description

Currently, there is only node verification without waiting for it to reach the ready status. 
Adding a wait for the node to be ready until timeout.

### CODE REVIEW

File: benchmark_runner/common/oc/oc.py [hunk 1]
```diff
@@ -457,32 +457,51 @@ def get_worker_nodes(self):
...
     @typechecked
-    def check_node_status(nodes_list: list):
+    def wait_for_nodes_ready(self, wait_time: int = None, timeout: int = int(environment_variables.environment_variables_dict['timeout'])):
         """
-        This method check node status
-        @param nodes_list:
-        @return: True when all nodes in ready status
+        This method waits until all nodes are in 'Ready' status.
+        @param wait_time: wait time between each loop
+        @param timeout: Maximum wait time in seconds (default set in environment variables)
```
Comment (RobertKrawitz) [inline benchmark_runner/common/oc/oc.py:465]: Need to document that negative value means no timeout (which I think is a good thing to have).

File: benchmark_runner/common/oc/oc.py [hunk 2]
```diff
@@ -457,32 +457,51 @@ def get_worker_nodes(self):
...
+        This method checks the status of all nodes in the list.
+        @return: True if all nodes are in 'Ready' status; otherwise, return (node_name, node_status)
         """
         # Check if any node is not in 'Ready' status
-        for node in nodes_list:
+        for node in self.get_node_status():
             node_name, node_status = node.split()
             if node_status != 'Ready':
-                raise NodeNotReady(node_name, node_status)
+                return node_name, node_status
```
Comment (RobertKrawitz) [inline benchmark_runner/common/oc/oc.py:492]: Return all nodes that are not ready, not just the first one.

**Review by RobertKrawitz — CHANGES_REQUESTED** [review]
One nit.

File: benchmark_runner/common/oc/oc.py (hunk 2 above)
Comment (ebattat) [reply benchmark_runner/common/oc/oc.py]: Good catch, add dictionary to hold not ready node: `not_ready_nodes = {}`

File: benchmark_runner/common/oc/oc.py (hunk 1 above)
Comment (ebattat) [reply benchmark_runner/common/oc/oc.py]: Add it to the document

**Review by RobertKrawitz — APPROVED** [review]

**Pull request merged**